<?xml version="1.0" encoding="utf-8"?>
<launch>
  <include file="$(find pses_simulation)/launch/simulation.launch">
  	<arg name="rviz" value="true"/>
  	<arg name="dashboard" value="false"/>
  </include>

  <!-- Dashboard for gamepad -->
  <node pkg="autu_control" type="autu_dashboard" name="autu_dashboard" output="screen"/>
  <!-- Joystick driver -->
  <node pkg="joy" type="joy_node" name="joy_node" output="screen">
   	<param name="dev" value="/dev/input/js0"/>
   	<param name="max_motorlevel" value="20"/>
   	<param name="max_steering" value="35"/>
  </node>

  <node pkg="autu_control" type="main" name="main" output="screen">
    <remap from="autu/command" to="pses_basis/command"/>

    <param name="PDController/max_motor_level" value="20"/>
    <param name="PDController/p" value="10"/>
    <param name="PDController/d" value="1"/>

    <param name="curvedriver/max_motor_level" value="10"/>
    <param name="curvedriver/after_curve_deadtime" value="0.75"/>

    <param name="curvedriver_constant/max_motor_level" value="20"/>
    <param name="curvedriver_constant/steering" value="15"/>
    <param name="curvedriver_constant/precurve_distance" value="1.3"/>
    <param name="curvedriver_constant/corner_end_angle" value="60"/>
    <param name="curvedriver_constant/corner_threshold" value="0.4"/>
    <param name="curvedriver_constant/dist_corner_to_line" value="0.3"/>
    <param name="curvedriver_constant/false_corner_end" value="0.4"/>
    <param name="curvedriver_constant/blindness_offset" value="0.5"/>
    <param name="curvedriver_constant/radius" value="1.25"/>
    <param name="curvedriver_constant/corner_safety_distance" value="0.15"/>
  </node>
</launch>
